spot_rl_experiments/spot_rl/skills/atomic_skills.py

```diff
@@ -73,92 +76,67 @@ def __init__(
         self.policy.reset()
 
         return observations
 
-    def execute_nav(
-        self, nav_targets_list: List[Tuple[float, float, float]]
-    ) -> List[List[Dict]]:
+    def execute_rl_loop(self, goal_dict: Dict[str, Any]):
         """
-        @INFO: This function is only to be used locally in spot_rl_experiments
-        Executes the navigation to the given nav_targets_list and returns the robot's trajectory
+        !!BLOCKING CALL!!
+
+        Executes the action based on output from policy
+
         Args:
-            nav_targets_list: List of nav_targets (x,y,theta) where robot needs to navigate to
+            goal_dict: Dict[str, Any] containing necessary goal information for the appropriate skill
+        """
+        observations = self.reset_env_and_policy(goal_dict)
+        done = False
 
-        Returns:
-            robot_trajectories: [[Dict]] where each Dict contains timestamp and pose of the robot, inner list contains trajectory for each nav_target and outer list is a collection of each of the nav_target's trajectory
-        """
-
-        robot_trajectories = []  # type: List[List[Dict]]
-        for nav_target in nav_targets_list:
-            observations = self.reset_env_and_policy(nav_target)
-            done = False
-
-            # List of Dicts to store trajectory for
-            # each of the nav_targets in nav_targets_list
-            robot_trajectory = []  # type: List[Dict]
-            time.sleep(1)
-
-            self.env.say(f"Navigating to {nav_target}")
-
-            if self.record_robot_trajectories and not self.recording_in_progress:
-                self.recording_in_progress = True
-                # Set start time for recording before execution of 1st nav waypoint
-                start_time = time.time()
-
-            # Execution Loop
-            while not done:
-                action = self.policy.act(observations)
-                observations, _, done, _ = self.env.step(base_action=action)
-
-                # Record trajectories at every step if True
-                if self.record_robot_trajectories:
-                    robot_trajectory.append(
-                        {
-                            "timestamp": time.time() - start_time,
-                            "pose": [
-                                self.env.x,
-                                self.env.y,
-                                np.rad2deg(self.env.yaw),
-                            ],
-                        }
-                    )
-            # Store the trajectory for each nav_target inside the List[robot_trajectory]
-            if self.record_robot_trajectories:
-                robot_trajectories.append(robot_trajectory)
-
-        # Turn off recording
-        self.recording_in_progress = False
-
-        # Return waypoints back
-        return robot_trajectories
+        # Update logged data at init
+        self.do_stuff_pre_loop(goal_dict)
```

Comment:
Hi Kavit! Sorry for the nit-picking. I suggest we might want to use the following names
`self.do_stuff_pre_loop `-> `self.pre_execution_rl_loop`
`self.do_stuff_in_loop` -> `self.in_execution_rl_loop`
`self.do_stuff_post_loop `-> `self.post_execution_rl_loop`

Reply:
Yes, I wanted more suggestions on the name here.
@zephirefaith @akshararai

Reply:
addressed